src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -159,16 +159,35 @@ private BoundExpression VisitCollectionBuilderCollectionExpression(BoundCollecti
             Debug.Assert(!_inExpressionLambda);
             Debug.Assert(node.Type is { });
 
+            var syntax = node.Syntax;
+            var elements = node.Elements;
             var constructMethod = node.CollectionBuilderMethod;
+
             Debug.Assert(constructMethod is { });
             Debug.Assert(constructMethod.ReturnType.Equals(node.Type, TypeCompareKind.AllIgnoreOptions));
 
             var spanType = (NamedTypeSymbol)constructMethod.Parameters[0].Type;
             Debug.Assert(spanType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions));
 
-            var span = VisitArrayOrSpanCollectionExpression(node, spanType, spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0].Type);
-            return new BoundCall(
-                node.Syntax,
+            var elementType = spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0];
+            var locals = ArrayBuilder<LocalSymbol>.GetInstance();
+            var sideEffects = ArrayBuilder<BoundExpression>.GetInstance();
+            BoundExpression span;
+
+            // PROTOTYPE: Check also ReadOnlySpan<T> parameter is not [UnscopedRef].
+            if (elements.Length > 0
```

Comment:
Should there be any kind of upper limit/heuristic?

Reply:
For now, we don't think there should be a limit, because the elements are already explicit in the source and because the work around to use the heap is to use C#11 syntax.